### PR TITLE
Add SDL_copy() SDL_copyp() (see #5786)

### PR DESCRIPTION
**Pull request by 1bsyl**
Add SDL_copy() SDL_copyp() (see #5786)

### CODE REVIEW

File: src/video/os2/SDL_os2video.c [hunk 1]
```diff
@@ -1575,7 +1575,7 @@ static void OS2_GetDisplayModes(_THIS, SDL_VideoDisplay *display)
     SDL_copy(mode, display->current_mode);
     mode.driverdata = (MODEDATA *) SDL_malloc(sizeof(MODEDATA));
     if (!mode.driverdata) return; /* yikes.. */
-    SDL_copyp(mode.driverdata, display->current_mode.driverdata);
+    SDL_memcpy(mode.driverdata, display->current_mode.driverdata, sizeof(MODEDATA));
```
Comment (sezero) [inline src/video/os2/SDL_os2video.c:1578]: Yes: `current_mode.driverdata` has `void*` type, so the compiler rightfully rejects `sizeof(void)`

**Comment by slouken** [comment]
@1bsyl, can you resolve conflicts and recreate the PR as a single working commit?

**Comment by 1bsyl** [comment]
@slouken 
the conflict is resolved but I haven't much tested except than linux.
there is the "squash and merge" to have a single commit for all

File: src/video/os2/SDL_os2video.c (hunk 1 above)
Comment (1bsyl) [reply src/video/os2/SDL_os2video.c:1578]: yes I have seen that :)

**Comment by slouken** [comment]
Looking at the changes you made, I think we just want SDL_copyp as a direct replacement for memcpy, with the size check.

**Comment by 1bsyl** [comment]
@slouken 
if we only keep SDL_copyp, then it will be clear to check the diffs : only replacement of SDL_memcpy and remove last parameter.

Just confirmation: should we keep only SDL_copyp ?

**Comment by slouken** [comment]
> @slouken if we only keep SDL_copyp, then it will be clear to check the diffs : only replacement of SDL_memcpy and remove last parameter.
> 
> Just confirmation: should we keep only SDL_copyp ?

Yes please.

**Comment by sezero** [comment]
As far as can see, this is useful for types that can have confusing
variants, like SDL_Rect or SDL_Point:  Its use should be restricted
to those and such - e.g. its use in SDL_audio.c isn't really needed.
And it possibly needn't be made public either.

**Comment by 1bsyl** [comment]
indeed, should this be public or not ? 
Since there is SDL_zero{,a,p}, why not ... 
but we may also choose not to increase public API

**Pull request closed without merging**